app/moves/middleware.js

```diff
@@ -11,28 +13,44 @@ const moveService = require('../../common/services/move')
 const { LOCATIONS_BATCH_SIZE } = require('../../config')
 
 const moveDateFormat = 'yyyy-MM-dd'
+const weekStartsOn = 1
 
-function makeMultipleRequests(service, moveDate, locationIdBatches) {
+function makeMultipleRequests(service, dateRange, locationIdBatches) {
   return Promise.all(
-    locationIdBatches.map(chunk => service({ moveDate, fromLocationId: chunk }))
+    locationIdBatches.map(chunk =>
+      service({ dateRange, fromLocationId: chunk })
+    )
   )
 }
 
+function getPeriod(date, interval) {
+  const method = interval >= 0 ? addDays : subDays
+  return format(method(parseISO(date), Math.abs(interval)), moveDateFormat)
+}
+
 module.exports = {
   redirectBaseUrl: (req, res) => {
     const today = format(new Date(), moveDateFormat)
     const currentLocation = get(req.session, 'currentLocation.id')
 
     if (currentLocation) {
-      return res.redirect(`${req.baseUrl}/${today}/${currentLocation}`)
+      return res.redirect(`${req.baseUrl}/day/${today}/${currentLocation}`)
     }
 
-    return res.redirect(`${req.baseUrl}/${today}`)
+    return res.redirect(`${req.baseUrl}/day/${today}`)
   },
   saveUrl: (req, res, next) => {
     req.session.movesUrl = req.originalUrl
     next()
   },
+  setPeriod: (req, res, next, period) => {
+    res.locals.period = period
+    next()
+  },
+  setStatus: (req, res, next, status) => {
```

Comment:
We can probable remove this middleware. We don't use this value in the view and the subsequent middleware that currently accesses `res.locals.status` could access `req.params.status` to get the same value.